### PR TITLE
Propagate the merge result from graveler to the controller level

### PR DESCRIPTION
**Pull request by nopcoder**
Part of #2699

### CODE REVIEW

File: pkg/db/database.go [hunk 1]
```diff
@@ -190,29 +190,29 @@ func (d *PgxDatabase) Transact(ctx context.Context, fn TxFunc, opts ...TxOpt) (i
 			rollbackErr := tx.Rollback(ctx)
 			if rollbackErr != nil {
 				// returning the original error and not the rollbackErr
-				return nil, err
+				return ret, err
 			}
 			// retry on serialization error
-			if IsSerializationError(err) {
+			if IsSerializationError(rollbackErr) {
```
Comment (guy-har) [inline pkg/db/database.go:196]:
```suggestion
			if IsSerializationError(err) {
```

File: pkg/db/database.go [hunk 2]
```diff
@@ -190,29 +190,29 @@ func (d *PgxDatabase) Transact(ctx context.Context, fn TxFunc, opts ...TxOpt) (i
 			rollbackErr := tx.Rollback(ctx)
 			if rollbackErr != nil {
 				// returning the original error and not the rollbackErr
-				return nil, err
+				return ret, err
```
Comment (guy-har) [inline pkg/db/database.go:193]: Add a comment explaining the reason for returning ret

File: pkg/api/controller.go [hunk 3]
```diff
@@ -2716,7 +2716,7 @@ func (c *Controller) MergeIntoBranch(w http.ResponseWriter, r *http.Request, bod
 		writeError(w, http.StatusPreconditionFailed, err)
 		return
 	case errors.Is(err, catalog.ErrConflictFound) || errors.Is(err, graveler.ErrConflictFound):
-		writeError(w, http.StatusConflict, newMergeResultFromCatalog(res))
```
Comment (guy-har) [inline pkg/api/controller.go:2719]: Consider checking if we can change the writeError signature to work only on errors

**Review by guy-har — APPROVED** [review]

File: pkg/api/controller.go (hunk 3 above)
Comment (nopcoder) [reply pkg/api/controller.go:2719]: Checked this option - but there are many cases that we serialize the different messages and not errors when we send a status code.

**Pull request merged**